cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -1,77 +1,573 @@
+- For **Standalone CDAP:** as containers are replaced with Java threads, a single
+  ``logback.xml`` file, located in  ``<cdap-sdk-home>/conf``, configures both "container"
+  and CDAP system service logging.
+  
+You can also use a custom "logback" file with your application, as described in the
+Developers' Manual section :ref:`application-logback`.
+
+Changing Program Log Levels
+---------------------------
+The CDAP :ref:`Logging HTTP RESTful API
+<http-restful-api-logging-changing-program-log-levels>` can be used to set the log levels
+of a program, either before a particular run or while it is running. Once changed, they
+can be reset back to what they were originally by using the :ref:`reset endpoint
+<http-restful-api-logging-resetting>`.
+
+- To configure the log level before an application starts, you can add the logger name as the
+  key and log level as the value in the preferences. The logger name should be prefixed with
+  ``system.log.level``. 
+
+  For example, if you want to set the log level of the ``HelloWorld`` class of the
+  :ref:`Hello World example <examples-hello-world>` to ``DEBUG``, you would use
+  ``system.log.level.co.cask.cdap.examples.helloworld.HelloWorld`` as the key and ``DEBUG``
+  as the value. This can be applied to any package or classes. If the logger name is
+  omitted, it will change the log level of ROOT.
+
+- To configure the log level of a program dynamically (such as a worker, flow, or service
+  which is currently running), see the :ref:`Logging HTTP RESTful API
+  <http-restful-api-logging-changing-program-log-levels>`.
+
+**Note:** The :ref:`Logging HTTP RESTful API
+<http-restful-api-logging-changing-program-log-levels>` for changing program log levels
+can only be used with programs that are run under Distributed CDAP. For changing the log
+levels of programs run under Standalone CDAP, you either modify the ``logback.xml`` file,
+or you :ref:`provide a "logback.xml" <application-logback>` with your application
+before it is deployed.
+
+
+.. _logging-monitoring-system-services-logs:
+
+CDAP System Services Logs
+=========================
+As CDAP system services run either on cluster edge nodes or in YARN containers, their
+logging and :ref:`its configuration <logging-monitoring-system-services-logs-configuration>` 
+depends on the service and where it is located.
+
+Retrieving Log Messages from a System Service
+---------------------------------------------
+The log messages emitted by CDAP system services can be retrieved by:
+
+- Using the :ref:`restful-api`: the :ref:`Logging HTTP RESTful API
+  <http-restful-api-logging>` details :ref:`downloading the logs
+  <http-restful-api-logging-downloading-system-logs>` emitted by a system service.
+- Log messages of system services can be viewed in the :ref:`CDAP UI <cdap-ui>`.
+
+- Log messages of a program can be viewed in the :ref:`CDAP UI <cdap-ui>`. In the
```

Comment:
Fixed. Bad copy and paste. That sentence was suppose to be replaced by line 219.